Jenkinsfile

```diff
@@ -0,0 +1,127 @@
+@Library('common-shared') _
+
+pipeline {
+  agent {
+    kubernetes {
+      label 'kubedeploy-agent'
+      yaml '''
+      apiVersion: v1
+      kind: Pod
+      spec:
+        containers:
+        - name: kubectl
+          image: eclipsefdn/kubectl:1.9-alpine
+          command:
+          - cat
+          tty: true
+      '''
+    }
+  }
+
+  environment {
+    APP_NAME = 'geoip-rest-api'
+    NAMESPACE = 'foundation-internal-webdev-apps'
+    IMAGE_NAME = 'eclipsefdn/geoip-rest-api'
+    CONTAINER_NAME = 'geoip-rest-api'
+    ENVIRONMENT = sh(
+      script: """
+        if [ "${env.BRANCH_NAME}" = "master" ]; then
+          printf "production"
+        else
+          printf "${env.BRANCH_NAME}"
+        fi
+      """,
+      returnStdout: true
+    )
+    TAG_NAME = sh(
+      script: """
+        GIT_COMMIT_SHORT=\$(git rev-parse --short ${env.GIT_COMMIT})
+        if [ "${env.ENVIRONMENT}" = "" ]; then
+          printf \${GIT_COMMIT_SHORT}-${env.BUILD_NUMBER}
+        else
+          printf ${env.ENVIRONMENT}-\${GIT_COMMIT_SHORT}-${env.BUILD_NUMBER}
+        fi
+      """,
+      returnStdout: true
+    )
+  }
+
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '10'))
+  }
+
+  triggers { 
+    // build once a week to keep up with parents images updates
+    cron('H H * * H') 
+  }
+
+  stages {
+    stage('Build docker image') {
+      agent {
+        label 'docker-build'
+      }
+      steps {
+        sh '''
+          docker build -f src/main/docker/Dockerfile.jvm --no-cache -t ${IMAGE_NAME}:${TAG_NAME} -t ${IMAGE_NAME}:latest .
```

Comment:
the app needs to be built before the docker image can be assembled. You will need to add first:

`./mvnw package`